arelle/plugin/validate/ESEF/Dimensions.py

```diff
@@ -31,87 +35,90 @@ def addDomMbrs(sourceDomMbr, ELR, membersSet):
-        if i:
-            val.modelXbrl.warning("ESEF.3.4.2.extensionTaxonomyLineItemNotLinkedToAnyHypercube",
-                _("Dimensional line item reported non-dimensionally SHOULD be linked to \"not dimensionally qualified\" hypercube %(linkrole)s, primary item %(qnames)s"),
-                modelObject=i, linkrole=LineItemsNotQualifiedLinkrole, qnames=", ".join(sorted(str(c.qname) for c in i)))
-        # pri items in LineItemsNotQualifiedLinkrole which are not used in report non-dimensionally
-        # check no longer in Filer Manual as of 2021
-        #i = set(hcPrimaryItem
-        #       for hcPrimaryItem in elrPrimaryItems.get(LineItemsNotQualifiedLinkrole, set())
-        #       if not any(not f.context.qnameDims 
-        #                  for f in val.modelXbrl.factsByQname.get(hcPrimaryItem.qname,())
-        #                  if f.context is not None))
-        #if i:
-        #    val.modelXbrl.warning("ESEF.3.4.2.extensionTaxonomyLineItemIncorrectlyLinkedToNonDimensionallyQualifiedHypercube",
-        #        _("Dimensional line item not reported non-dimensionally has no need to be linked to \"not dimensionally qualified\" hypercube %(linkrole)s, primary item %(qnames)s"),
-        #        modelObject=i, linkrole=LineItemsNotQualifiedLinkrole, qnames=", ".join(sorted(str(c.qname) for c in i)))
+    for hasHypercubeArcrole in (XbrlConst.all, XbrlConst.notAll):
+        hasHypercubeRelationships = val.modelXbrl.relationshipSet(hasHypercubeArcrole).fromModelObjects()
+        for hasHcRels in hasHypercubeRelationships.values():
+            for hasHcRel in hasHcRels:
+                sourceConcept = hasHcRel.fromModelObject
+                hcPrimaryItems.add(sourceConcept)
+                # find associated primary items to source concept
+                for domMbrRel in val.modelXbrl.relationshipSet(XbrlConst.domainMember).fromModelObject(sourceConcept):
+                    if domMbrRel.consecutiveLinkrole == hasHcRel.linkrole: # only those related to this hc
+                        addDomMbrs(domMbrRel.toModelObject, domMbrRel.consecutiveLinkrole, hcPrimaryItems)
+                val.primaryItems.update(hcPrimaryItems)
+                hc = hasHcRel.toModelObject
+                if hasHypercubeArcrole == XbrlConst.all:
+                    if not hasHcRel.isClosed and isExtension(val, hasHcRel):
+                        val.modelXbrl.error("ESEF.3.4.2.openPositiveHypercubeInDefinitionLinkbase",
+                            _("Hypercubes appearing as target of definition arc with http://xbrl.org/int/dim/arcrole/all arcrole MUST have xbrldt:closed attribute set to \"true\""
+                              ": hypercube %(hypercube)s, linkrole %(linkrole)s, primary item %(primaryItem)s"),
+                            modelObject=hasHcRel, hypercube=hc.qname, linkrole=hasHcRel.linkrole, primaryItem=sourceConcept.qname)
+                elif hasHypercubeArcrole == XbrlConst.notAll:
```

Comment:
NIt: This check and the above check can be smashed and you can just pass `hasHypercubeArcrole`